tools/misc/labelme2coco.py

```diff
@@ -0,0 +1,322 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import argparse
+import glob
+import json
+import os
+import shutil
+
+import numpy as np
+from sklearn.model_selection import train_test_split
+from tqdm import tqdm
+
+# import sys
+
+
+class Labelme2coco_keypoints():
+
+    def __init__(self, args):
+        """Lableme 关键点数据集转 COCO 数据集的构造函数:
+
+        Args
+            args：命令行输入的参数
+                - class_name 根类名字
+        """
+
+        self.classname_to_id = {args.class_name: 1}
+        self.images = []
+        self.annotations = []
+        self.categories = []
+        self.ann_id = 0
+        self.img_id = 0
+
+    def save_coco_json(self, instance, save_path):
+        json.dump(
+            instance,
+            open(save_path, 'w', encoding='utf-8'),
+            ensure_ascii=False,
+            indent=1)
+
+    def read_jsonfile(self, path):
+        with open(path, 'r', encoding='utf-8') as f:
+            return json.load(f)
+
+    def _get_box(self, points):
+        min_x = min_y = np.inf
+        max_x = max_y = 0
+        for x, y in points:
+            min_x = min(min_x, x)
+            min_y = min(min_y, y)
+            max_x = max(max_x, x)
+            max_y = max(max_y, y)
+        return [min_x, min_y, max_x - min_x, max_y - min_y]
+
+    def _get_keypoints(self, points, keypoints, num_keypoints, label):
+        """解析 labelme 的原始数据， 生成 coco 标注的 关键点对象.
+
+        例如：
+            "keypoints": [
+                67.06149888292556,  # x 的值
+                122.5043507571318,  # y 的值
+                1,                  # 相当于 Z 值，2D关键点 v = 0表示不可见,
+                                      v = 1表示标记但不可见，v = 2表示标记且可见
+                82.42582269256718,
+                109.95672933232304,
+                1,
+                ...,
+            ],
+        """
+        labels = ['wrist', 'thumb1', 'thumb2', ...]
+        flag = label.split('_')[-1]
```

Comment:
There is a `flag` entry in Labelme format, consider using that?

Reply:
About this `flag`, I use it to distinguish from occluded or unoccluded keypoints because of the coco format 'keypoint = [x,y,v,....]'.
this v means visible for occluding

Reply:
> About this `flag`, I use it to distinguish from occluded or unoccluded keypoints because of the coco format 'keypoint = [x,y,v,....]'. this v means visible for occluding

Each point in Labelme format has a `flags` field for you to add flags, and you can use it to distinguish the occluded keypoints, it is better than writing all flags in the label.

Reply:
> > About this `flag`, I use it to distinguish from occluded or unoccluded keypoints because of the coco format 'keypoint = [x,y,v,....]'. this v means visible for occluding
> 
> Each point in Labelme format has a `flags` field for you to add flags, and you can use it to distinguish the occluded keypoints, it is better than writing all flags in the label.

I didn't understand the role of the flag before,thanks.